### PR TITLE
Update collection TS definitions

### PR DESCRIPTION
**Pull request by adc-mhaugen**
Add missing method TS definitions for Collection object.

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -88,9 +88,47 @@ declare module "miragejs" {
...
+    save(): Collection<T>;
+
+    /** Returns a new Collection with a subset of its models selected from begin to end. */
+    slice(begin: number, end: number): Collection<T>;
+
+    /** Returns a new Collection with its models sorted according to the provided compare function. */
+    sort(f: (a: T, b: T) => number): Collection<T>;
+
+    /** Updates each model in the collection, and immediately persists all changes to the db. */
+    update: <K extends keyof T>(key: K, val: T[K]) => Collection<T>;
```
Comment (dfreeman) [inline types/index.d.ts:131]: [nit] can you use method syntax for this? There are some subtleties with how subtyping behaves based on how function/method fields are declared.

```suggestion
    update<K extends keyof T>(key: K, val: T[K]): Collection<T>;
```

**Review by dfreeman — APPROVED** [review]
Thank you, @adc-mhaugen!

File: types/tests/server-test.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ new Server({
   routes() {
     this.namespace = "api";
 
-    this.schema.all("asfd"); // $ExpectType Collection<ModelInstance<{}>>
+    this.schema.all("asfd"); // $ExpectType Collection<Instantiate<Registry<AnyModels, AnyFactories>, "asfd">>
```
Comment (adc-mhaugen) [inline types/tests/server-test.ts:67]: @dfreeman Looks like the expected type here changed.  Is this correct?

Comment (adc-mhaugen) [reply types/tests/server-test.ts:67]: Hmmm, the test are now failing with this change, so perhaps I was too hasty.

**Comment by IanVS** [comment]
@adc-mhaugen we've updated our CI infrastructure, would you be willing to rebase this PR on master?  I'd like to get it green and merged in, if possible.

**Comment by IanVS** [comment]
Merging from upstream was all that was needed, especially since I squash-merged.  If we needed to keep the commits, and wanted a cleaner history, an actual rebase would be done by running `git refresh && git rebase upstream/master`.  But all I wanted was for CI to run, so merging master worked fine too.  :+1:

**Pull request merged**